internal/factsengine/gatherers/hostsfile.go

```diff
@@ -0,0 +1,144 @@
+package gatherers
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/internal/factsengine/entities"
+)
+
+const (
+	HostsFileFactKey    = "hosts"
+	HostsFilePath       = "/etc/hosts"
+	ipMatchGroup        = "ip"
+	hostnamesMatchGroup = "hostnames"
+	parsingRegexp       = `(?m)(?P<` + ipMatchGroup + `>\S+)\s+(?P<` + hostnamesMatchGroup + `>.+)`
+)
+
+var (
+	hostsEntryCompiled = regexp.MustCompile(parsingRegexp)
+)
+
+// nolint:gochecknoglobals
+var (
+	HostsFileError = entities.FactGatheringError{
+		Type:    "hosts-file-error",
+		Message: "error reading /etc/hosts file",
+	}
+
+	HostsFileDecodingError = entities.FactGatheringError{
+		Type:    "hosts-file-decoding-error",
+		Message: "error decoding /etc/hosts file",
+	}
+
+	HostsEntryNotFoundError = entities.FactGatheringError{
+		Type:    "hosts-file-value-not-found",
+		Message: "requested field value not found in /etc/hosts file",
+	}
+)
+
+type HostsFileGatherer struct {
+	hostsFile string
```

Comment:
I think that in general we could and we should use afero when possibile, in that case I think we could drop it, we could inject the path and use something like a fixture which we are doing right know in tests.

I just suggest to rename the hostsFile parameter here and in constructor function to hostsFilePath, is less misleading and we know right that it's the path and not the file content.

Reply:
+1 to rename this prior merging

Reply:
Aight :+1: